src/main/scala/org/apache/spark/ml/scaladl/MultilayerPerceptronRegressor.scala

```diff
@@ -0,0 +1,409 @@
+
+  setDefault(minimum -> 0.0, maximum -> 0.0)
+}
+
+/** Label to vector converter. */
+private object RegressionLabelConverter {
+
+  /**
+   * Encodes a label as a vector.
+   * Returns a vector of length 1 with the label in the 0th position
+   *
+   * @param labeledPoint labeled point
+   * @return pair of features and vector encoding of a label
+   */
+  def encodeLabeledPoint(labeledPoint: LabeledPoint, min: Double, max: Double): (Vector, Vector) = {
+    val output = Array.fill(1)(0.0)
+    if (max-min != 0.0) {
+      output(0) = (labeledPoint.label - min) / (max - min)
+    }
+    else {
+    // When min and max are equal, cannot min-max scale due to divide by zero error. Setting scaled
+    // result to zero will lead to consistent predictions, as the min will be added during decoding.
+    // Min and max will both be 0 if label scaling is turned off, and this code branch will run.
+      output(0) = labeledPoint.label - min
+    }
+    (labeledPoint.features, Vectors.dense(output))
+  }
+
+  /**
+   * Converts a vector to a label.
+   * Returns the value of the 0th element of the output vector.
+   *
+   * @param output label encoded with a vector
+   * @return label
+   */
+  def decodeLabel(output: Vector, min: Double, max: Double): Double = {
+    if (max-min != 0.0) {
+      (output(0) * (max - min)) + min
+    } else {
+      output(0)
+    }
+  }
+}
+
+ /**
+  * :: Experimental ::
+  * Regression trainer based on Multi-layer perceptron regression.
+  * Contains sigmoid activation function on all layers, output layer has a linear function.
+  * Number of inputs has to be equal to the size of feature vectors.
+  * Number of outputs has to be equal to one.
+  */
+class MultilayerPerceptronRegressor (
+    override val uid: String)
+  extends Predictor[Vector, MultilayerPerceptronRegressor, MultilayerPerceptronRegressorModel]
+    with MultilayerPerceptronBasicParams with MultilayerPerceptronRegressorParams with Serializable
+    with DefaultParamsWritable {
+
+  /**
+   * Sets the value of param [[initialWeights]].
+   *
+   * @group expertSetParam
+   */
+  def setInitialWeights(value: Vector): this.type = set(initialWeights, value)
+
+  /**
+   * Sets the value of param [[solver]].
+   * Default is "l-bfgs".
+   *
+   * @group expertSetParam
+   */
+  def setSolver(value: String): this.type = set(solver, value)
+
+  /**
+   * Sets the value of param [[stepSize]] (applicable only for solver "gd").
+   * Default is 0.03.
+   *
+   * @group setParam
+   */
+  def setStepSize(value: Double): this.type = set(stepSize, value)
+
+  def this() = this(Identifiable.randomUID("mlpr"))
+
+  override def copy(extra: ParamMap): MultilayerPerceptronRegressor = defaultCopy(extra)
+
+  /**
+   * Train a model using the given dataset and parameters.
+   *
+   * @param dataset Training dataset
+   * @return Fitted model
+   */
+  override protected def train(dataset: Dataset[_]): MultilayerPerceptronRegressorModel = {
```

Comment:
We should support multiple outputs, i.e. `dataset` should contain two vector columns - `input` and `output`.